vault/expiration.go

```diff
@@ -114,6 +119,91 @@ func (c *Core) stopExpiration() error {
 	return nil
 }
 
+// Tidy cleans up stale leases which are associated with invalid tokens
+func (m *ExpirationManager) Tidy() error {
+	var tidyErrors *multierror.Error
+
+	// Create a cache to keep track of looked up tokens
+	tokenCache := make(map[string]string)
```

Comment:
You don't need to store all tokens you have seen; you only need to store a list of _valid_ tokens so that you don't check for validity again. So treat it as a set via `map[string]struct{}`.